hermes/crates/cbork/cddl-parser/tests/cddl.rs

```diff
@@ -0,0 +1,56 @@
+use std::{ffi::OsStr, fs, io::Result};
+
+use cddl_parser::{parse_cddl, Extension};
+
+#[test]
+/// # Panics
+fn parse_cddl_files() {
+    let entries = fs::read_dir("tests/cddl").expect("`tests/cddl` directory must exist");
+
+    let mut file_paths: Vec<_> = entries
+        .filter_map(Result::ok)
+        .filter_map(|x| x.path().is_file().then_some(x.path()))
+        .collect();
+
+    file_paths.sort();
+
+    let valid_file_paths = file_paths.iter().filter(|p| {
+        matches!(
+            p.file_name()
+                .and_then(OsStr::to_str)
+                .map(|p| p.starts_with("valid")),
+            Some(true)
+        )
+    });
+    let invalid_file_paths = file_paths.iter().filter(|p| {
+        matches!(
+            p.file_name()
+                .and_then(OsStr::to_str)
+                .map(|p| p.starts_with("invalid")),
+            Some(true)
+        )
+    });
+
+    // test for valid files
+    let mut err_messages = vec![];
+    for file_path in valid_file_paths {
+        let mut content = fs::read_to_string(file_path).expect("failed to read a file");
+
+        if let Err(e) = parse_cddl(&mut content, &Extension::RFC8610Parser) {
```

Comment:
```suggestion
        if let Err(e) = parse_cddl(&mut content, &Extension:CDDLParser) {
```
This should be using the CDDL Parser, , as the Fully featured parser should work with all CDDL file types.